src/pool/poolClient.ts

```diff
@@ -125,49 +126,50 @@ type EventIdParams = { blockNumber: number; transactionIndex: number; logIndex:
 export class PoolEventState {
   private seen = new Set<string>();
   private iface: ethers.utils.Interface;
-  constructor(
-    private contract: hubPool.Instance,
-    private startBlock = 0,
-    private state: hubPool.EventState = hubPool.eventStateDefaults()
-  ) {
+  // maintain ordered unique list of events so we can calculate state
+  private events: uma.SerializableEvent[] = [];
```

Comment:
The logic in this class has changed a lot. It previously assumed all events were queried, such that we could optimize by continuously updating the last block seen, and only storing the final state from the events.  Now that each user is queried separately, we basically query from the earliest block every time, then just storing the events, and rebuilding state from the full list of events every time.  We have multiple sources of events, from provider queries, and transaction receipts, so these are all merged into the same list.